### ISSUE TITLE
Unable to sign in thru Harvest

### ISSUE THREAD

**Issue by carrchr** [Strategery-Inc/jira-harvest-time-tracker#108]
**Describe the bug**
Login pop-up window says  

**To Reproduce**
1. Go to any Jira issue
2. Click on Sign Into Harvest button
3. new Browser Window pops-up and displays Authorize sign in to Harvest
4. Nothing changes in Jira. See errors in browser console

**Expected behavior**
Login and be able to enter time

**Screenshots**
<img width="662" alt="Screen Shot 2021-12-13 at 3 58 30 PM" src="https://user-images.githubusercontent.com/1752939/145887993-278a3902-775f-46af-a6ea-8f880cb575b2.png">

**Browser Console Logs**
Access to fetch at 'https://<our domain>.harvestapp.com/v2/users/me?' from origin 'https://<our Jira server>' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
I also emailed you

**Please complete the following information:**
 - JIRA Type: Server
 - JIRA Version: 8.13.14
 - Browser Type: Chrome on macOS
 - Browser Version: Version 96.0.4664.93 (Official Build) (x86_64)

**Additional context**
We hadn't renewed the plugin for a while. Upgraded Jira Server version and had to do so.

**Comment by gsomoza** [comment]
Hi @carrchr, a few things that might help:
1. Could you please confirm you're running the latest version of the module? That's currently `1.0.28-SNAPSHOT` for JIRA Server.
2. Could you please check the URL in your JIRA configuration and make sure the `API Host` field is set to `api.harvestapp.com`?
2. Please also make sure you're using one of the NEW harvest OAuth2 applications (https://id.getharvest.com/developers). They recently decommissioned their old API which could be another reason why you're seeing this if you recently updated your module.

**Comment by carrchr** [comment]
1. can confirm we are running latest version
2. updating API Host to `api.harvestapp.com` fixed it!
We had already done 3.

**Comment by gsomoza** [comment]
Nice, glad that fixed it!

**Comment by gsomoza** [comment]
I'll reopen this issue in case it can help someone else

**Comment by shernandez-ctrlQS** [comment]
Hi Gabriel / @gsomoza,

We switched to the newer version of the plug-in but struggled a bit to figure out why it didn´t work anymore (very same issue reported here).
I think it makes sense displaying better that users need to change the Harvest API URL from their custom instance´s one to the generic public one. I only found it because I came here to GitHub to open a ticket and saw this one.
Works for us now, but we spent time configuring, uninstalling, installing, etc... because it wouldn´t work and we simply were not aware of the API URL change with 1.0.28-SNAPSHOT.

Best,
Sergio

**Comment by gsomoza** [comment]
We're working this week on a notification that will send users to a documentation page with the steps clearly explained. Now that Harvest removed the need to point to different API hosts, we will also remove this option from the configuration section, making sure user can't break their configs by accident anymore and automatically upgrading anyone who didn't take these steps yet. The step of manually configuring the OAuth application in Harvest will still have to be done manually for Jira Server users though.. unfortunately there's not much we can do about that.

**Comment by andrewmallis** [comment]
@gsomoza can you please post the link to the documentation page here?

**Comment by gsomoza** [comment]
@andrewmallis see here: https://github.com/Strategery-Inc/jira-harvest-time-tracker/wiki/Debugging-Issues-with-Login - we'll expand with more information / details on an as-needed basis (feel free to share any questions / feedback).
